### PR TITLE
Encryption

### PR DESCRIPTION
**Pull request by ViktorLitviakov8050**
Encryption

### CODE REVIEW

File: notes/login/cryptogtapher.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from cryptography.fernet import Fernet
+
+class Cryptographer:
+    def __init__(self, key):
+        self.cryptosuit  = Fernet(key)
```
Comment (zasviat) [inline notes/login/cryptogtapher.py]: it's better to rename to 'cryptosuite' or 'crypto_suite'

File: notes/login/cryptogtapher.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+    def encrypt(self, password) -> str:
```
Comment (zasviat) [inline notes/login/cryptogtapher.py]: image

this method returns bytes not str
Let's change the return value of the 'encrypt' method

File: notes/login/views.py [hunk 3]
```diff
@@ -2,20 +2,21 @@
 from .models import User 
 from django.http import JsonResponse
 from django.db import IntegrityError
-
+from .cryptogtapher import Cryptographer
```
Comment (zasviat) [inline notes/login/views.py]: here's the typo in module name
```suggestion
from .cryptographer import Cryptographer
```

**Review by zasviat — APPROVED** [review]
good work!

**Comment by zasviat** [comment]
Bug: InvalidToken Exception when there is request to get random note by token. Reproducable with different tokens
image

Reason: encrypted password was saved in bytes format but as a string
image

Fix: Cryptographer now returns decoded (str, not bytes) encrypted password so we save encrypted decoded password to the database

After fix: no exception, password is decoded successfully
image
image

**Comment by graveetone** [comment]
@ViktorLitviakov8050 we need to make a couple of changes in this PR:
1. Let's move cryptographer.py to notes/notes instead of storing it in login module. Cryptographer is used in both 'notes' and 'randomNote' apps.
2. As we have randomNote app in project, we need to specify its name in notes/settings.y to "let the Django know about it"

**Review by zasviat — CHANGES_REQUESTED** [review]
we need to make a couple of changes in this PR:

1. Let's move cryptographer.py to notes/notes instead of storing it in login module. Cryptographer is used in both 'notes' and 'randomNote' apps.

2. As we have randomNote app in project, we need to specify its name in notes/settings.y to "let the Django know about it"

**Pull request merged**